core/capabilities/triggers/logevent/trigger.go

```diff
@@ -0,0 +1,210 @@
+package logevent
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// Log Event Trigger Capability Request Config Details
+type RequestConfig struct {
+	ContractName      string `json:"contractName"`
+	ContractAddress   string `json:"contractAddress"`
+	ContractEventName string `json:"contractEventName"`
+	// Log Event Trigger capability takes in a []byte as ContractReaderConfig
+	// to not depend on evm ChainReaderConfig type and be chain agnostic
+	ContractReaderConfig map[string]any `json:"contractReaderConfig"`
+}
+
+// LogEventTrigger struct to listen for Contract events using ContractReader gRPC client
+// in a loop with a periodic delay of pollPeriod milliseconds, which is specified in
+// the job spec
+type logEventTrigger struct {
+	ch   chan<- capabilities.TriggerResponse
+	lggr logger.Logger
+
+	// Contract address and Event Signature to monitor for
+	reqConfig      *RequestConfig
+	contractReader types.ContractReader
+	relayer        core.Relayer
+	startBlockNum  uint64
+
+	// Log Event Trigger config with pollPeriod and lookbackBlocks
+	logEventConfig Config
+	ticker         *time.Ticker
+	stopChan       services.StopChan
+	done           chan bool
+}
+
+// Construct for logEventTrigger struct
+func newLogEventTrigger(ctx context.Context,
+	lggr logger.Logger,
+	workflowID string,
+	reqConfig *RequestConfig,
+	logEventConfig Config,
+	relayer core.Relayer) (*logEventTrigger, chan capabilities.TriggerResponse, error) {
+	jsonBytes, err := json.Marshal(reqConfig.ContractReaderConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create a New Contract Reader client, which brings a corresponding ContractReader gRPC service
+	// in Chainlink Core service
+	contractReader, err := relayer.NewContractReader(ctx, jsonBytes)
+	if err != nil {
+		return nil, nil,
+			fmt.Errorf("error fetching contractReader for chainID %s from relayerSet: %v", logEventConfig.ChainID, err)
```

Comment:
The linter is supposed to suggest using `%w` for errors:
```suggestion
			fmt.Errorf("error fetching contractReader for chainID %s from relayerSet: %w", logEventConfig.ChainID, err)
```

Reply:
Fixed this and all other occurences of the error %v